src/ccaligner.cpp

```diff
@@ -33,62 +34,36 @@ void printHeader(std::string version)
     std::cout<<"=====================================================\n\n";
 }
 
-void printFooter()
+static void printFooter()
 {
     std::cout<<"\n\n=====================================================\n";
     std::cout<<"Issues? Open a ticket here\n";
     std::cout<<"https://github.com/saurabhshri/CCAligner/issues\n";
 }
 
-CCAligner::CCAligner(Params * parameters)
+int main(int argc, char *argv[])
 {
-    _parameters = parameters;
-    should_log = _parameters->verbosity ;
-}
+    printHeader("0.03 Alpha [Shubham]");
 
-int CCAligner::initAligner()
-{
-    if(_parameters->chosenAlignerType == approxAligner)
+    Params parameters;
+    parameters.inputParams(argc,argv);
+
+    if(parameters.chosenAlignerType == approxAligner)
     {
-        ApproxAligner * aligner = new ApproxAligner(_parameters->subtitleFileName, srt);
```

Comment:
The reason I went for heap allocation is that the size of files we are dealing with could be huge. It makes sense to not keep them in stack, but I am open for suggestion.

Reply:
Ah, I understand. But it's not a good idea to leak implementation details like that. Also, `std::fstream` already uses heap allocations when reading files, so there is no need for your classes to have to be allocated on the heap. 

I couldn't find any information on this on the internet, but it makes sense: For `std::fstream` to be able to store the file in stack there will already have to be stack memory that it can use, i.e. like a `char Buffer[Size];` somewhere. If this exists, which I doubt BTW - never really seen this, then it's only for a small value of `Size`, as that memory will be allocated on the stack every time you create that object, thus wasting memory if `Size` is big. This is comparable to SSO and `std::function`'s behavior for small function objects. So, if you read a big file, it will be allocated on the heap, so no worries. :)